pygmt/tests/test_contour.py

```diff
@@ -107,3 +107,37 @@ def test_contour_from_file(region):
         data=POINTS_DATA, projection="X10c", region=region, frame="af", pen="#ffcb87"
     )
     return fig
+
+
+@pytest.mark.mpl_image_compare(filename="test_contour_vec.png")
+def test_contour_deprecate_columns_to_incols(region):
+    """
+    Make sure that the old parameter "columns" is supported and it reports an
+    warning.
+
+    Modified from the test_contour_vec() test.
+    """
+    fig = Figure()
+    x, y = np.meshgrid(
+        np.linspace(region[0], region[1]), np.linspace(region[2], region[3])
+    )
+    x = x.flatten()
+    y = y.flatten()
+    z = (x - 0.5 * (region[0] + region[1])) ** 2 + 4 * y ** 2
+    z = np.exp(-z / 10 ** 2 * np.log(2))
+
+    # generate dataframe
+    # switch x and y from here onwards to simulate different column order
+    data = np.array([y, x, z]).T
+
+    with pytest.warns(expected_warning=FutureWarning) as record:
+        fig.contour(
+            data=data,
+            projection="X10c",
+            region=region,
+            frame="a",
+            pen=True,
+            columns=[1, 0, 2],
+        )
+        assert len(record) == 1  # check that only one warning was raised
+    return fig
```

Comment:
One thing I'm wondering is if we need to add deprecation tests for all of these methods (`contour` here and `rose` in #1306) since it'll be a chore to remove them in v0.6.0. Maybe having them in `fig.plot` is enough?

Reply:
In principle I agree with you @weiji14. However, especially the new test added to `test_contour.py` fails and seems like the `incols`/`columns` parameter is ignored at all so far. Thus, in my opinion it makes sense to test the deprecated parameters.

Reply:
Ok, let's keep the test then. And about why it's failing, it seems that the plot is outside of the `region` ([10, 70, -5, 10]). Not sure if this is an upstream bug or 'feature', but the region seems to be parsed before the column positions are swapped by incols (`-i`)?:

| expected | actual (wrong?) |
|--|--|
| ![correct](https://user-images.githubusercontent.com/23487320/120142740-ea4cff00-c232-11eb-8f3a-580cade5a0e9.png) | ![incorrect](https://user-images.githubusercontent.com/23487320/120142745-ec16c280-c232-11eb-83ee-d46253c9f866.png) |

Reply:
Yes, the columns are not swapped although ``incols`` is correctly given to GMT via ``@GMTAPI@-S-I-D-M-T-N-000000 -Ba -JX10c -R10/70/-5/10 -W -i1,0,2``.

Reply:
Looks like an upstream bug to me. Could you please create a new issue with a minimal example to reproduce it (see https://github.com/GenericMappingTools/pygmt/issues/1309 for an example), so that Paul and others can decide if it's an upstream issue and debug it.

Reply:
Problem with -i parsing fixed upstream in https://github.com/GenericMappingTools/gmt/pull/5289, and the test passes now :tada: